src/resolution.rs

```diff
@@ -130,14 +133,33 @@ pub fn resolve(
             }
         }
 
-        // TODO: Pull PackageSource from our dependency request.
-        let default_registry = package_sources
-            .get(&PackageSourceId::DefaultRegistry)
-            .unwrap();
+        let mut candidates = None;
+        let mut source_registry = &PackageSourceId::DefaultRegistry;
 
-        // Pull all of the possible candidate versions of the package we're
-        // looking for from the index.
-        let mut candidates = default_registry.query(&dependency_request.package_req)?;
+        // Look through all our packages sources in order of priority
+        for package_source in &package_sources.source_order {
+            source_registry = package_source;
+            let registry = package_sources.get(&package_source).unwrap();
+
+            // Pull all of the possible candidate versions of the package we're
+            // looking for from this source.
+            match registry.query(&dependency_request.package_req) {
+                Ok(manifests) => {
+                    candidates = Some(manifests);
+                    break;
+                }
+                Err(_) => {}
+            };
+        }
+
+        // Check if any of the sources provided candidates, if they didn't we can't proceed
+        let mut candidates = match candidates {
+            Some(candidates) => candidates,
+            None => bail!(
+                "Failed to find a source for {}",
+                dependency_request.package_req
+            ),
+        };
```

Comment:
Can we replace this code with `package_sources.source_order.iter().find()`?

Reply:
Also much cleaner!